### PR TITLE
2532 optimize estimation for manager operations

### PR DESCRIPTION
**Pull request by hui-an-yang**
<dl>
  <dt> **[breaking change]** depcrecated hardcoded constants `DEFAULT_GAS_LIMIT`, `DEFAULT_STORAGE_LIMIT` and `DEFAULT_FEE`</dt>
<dd> - exposed new function to `getRevealFee(address)` replacing `DEFAULT_FEE.REVEAL` to get accurate reveal fee based on address type</dd>
 <dd> - exposed new constants `REVEAL_STORAGE_LIMIT ` replacing `DEFAULT_STORAGE_LIMIT.REVEAL` </dd>
 <dd> - exposed new constants `ORIGINATION_SIZE` replacing `DEFAULT_STORAGE_LIMIT. ORIGINATION ` & `DEFAULT_STORAGE_LIMIT.TRANSFER ` </dd>
</dl>

* reduced unrevealed address operations prerequisite reveal fee from fixed 375 to 276-305 depends on address type
* added logic to add buffer gas to gasLimit depends on the operations
* reduced suggestedFeeMutez buffer from 100 to 20 per operation and use more precise length to calculate fee
* Refined gasLimit and storageLimit according to simulate_operation results and octez-client

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have run the linter against the changes
- [ ] You have added unit tests (if relevant/appropriate)
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

In this PR, please also make sure: 

- [ ] You have linked this PR to the issue by putting `closes #TICKETNUMBER` in the description box (when applicable)
- [ ] You have added a concise description on your changes
## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: packages/taquito/src/operations/types.ts [hunk 1]
```diff
@@ -76,6 +76,27 @@ export type RPCOpWithSource =
   | RPCSmartRollupAddMessagesOperation
   | RPCSmartRollupOriginateOperation;
 
+export const isOpWithGasBuffer = <
```
Comment (dsawali) [inline packages/taquito/src/operations/types.ts:79]: do you mind explaining a bit what this method does?

File: packages/taquito/src/constants.ts [hunk 2]
```diff
@@ -88,19 +109,37 @@ export enum ChainIds {
...
+    case 'tz3':
+      return REVEAL_GAS_LIMIT.TZ3;
+    case 'tz4':
+      return REVEAL_GAS_LIMIT.TZ4;
+    default:
+      throw new Error(`Cannot estimate reveal gas limit for ${address}`);
+  }
+};
+
+export const getRevealFee = (address: string) => getRevealFeeInternal(address);
```
Comment (dsawali) [inline packages/taquito/src/constants.ts:131]: what's the reason that getRevealFee is layered on top of getRevealFeeInternal?

File: packages/taquito/src/operations/types.ts (hunk 1 above)
Comment (hui-an-yang) [reply packages/taquito/src/operations/types.ts:79]: from my research that octez-client only add 100 gas buffer to these operation `[update_consensus_key, register_global_constants, transfer_ticket, originate, contract_call, smart_rollup_originate, smart_rollup_add_message]`
gasLimit will correlated of fee calculation so being precise of which ops to add which doesn't help lower fee

File: packages/taquito/src/constants.ts (hunk 2 above)
Comment (hui-an-yang) [reply packages/taquito/src/constants.ts:131]: good call out, was trying to have same structure as `getRevealGasLimit` but getRevealFee doesn't add logic value will update

**Review by dsawali — APPROVED** [review]
great work!

**Pull request merged**